moped-editor/src/views/projects/newProjectView/NewProjectMap.js

```diff
@@ -200,11 +276,57 @@ const NewProjectMap = ({
       : "default";
   };
 
+  /**
+   * Whenever a Save Action is initiated, save all drawn features
+   * */
+  useEffect(() => {
+    // Ignore empty states
+    if (!saveActionState) return;
+    // If the process has been already initiated, we don't need to go any further
+    if (saveActionState?.currentStep && saveActionState.currentStep > 1) return;
+    // It looks like this is the first step
+    if (
+      saveActionState?.initiateFeatureSave &&
+      saveActionState?.featuresSaved === false
+    ) {
+      saveDrawnPoints();
+    }
+  }, [saveActionState, saveDrawnPoints]);
+
```

Comment:
This useEffect tracks the state of the reducer, it receives the signal to save all drawn points (initiateFeatureSave).